aptos-move/aptos-transaction-simulation/src/state_store.rs

```diff
@@ -0,0 +1,447 @@
+            Some(blob) => Ok(bcs::from_bytes(&blob)?),
+            None => Ok(None),
+        }
+    }
+    /// Modifies a Move resource stored at the specified address in-place.
+    ///
+    /// The Rust type representing the Move resource must implement the [`MoveResource`] trait.
+    fn modify_resource<R, F>(&self, addr: AccountAddress, modify: F) -> Result<()>
+    where
+        R: MoveResource + Serialize,
+        F: FnOnce(&mut R) -> Result<()>,
+    {
+        let mut resource = match self.get_resource(addr)? {
+            Some(resource) => resource,
+            None => bail!("failed to modify resource -- resource does not exist"),
+        };
+        modify(&mut resource)?;
+        self.set_resource(addr, &resource)
+    }
+
+    /// Sets an on-chain config.
+    ///
+    /// The Rust type representing the on-chain config must implement the [`OnChainConfig`] trait.
+    fn set_on_chain_config<C>(&self, config: &C) -> Result<()>
+    where
+        C: OnChainConfig + Serialize,
+    {
+        let addr = AccountAddress::from_hex_literal(C::ADDRESS).unwrap();
+
+        self.set_state_value(
+            StateKey::resource(&addr, &StructTag {
+                address: addr,
+                module: Identifier::new(C::MODULE_IDENTIFIER).unwrap(),
+                name: Identifier::new(C::TYPE_IDENTIFIER).unwrap(),
+                type_args: vec![],
+            })?,
+            StateValue::new_legacy(bcs::to_bytes(&config)?.into()),
+        )
+    }
+    /// Gets an on-chain config.
+    ///
+    /// The Rust type representing the on-chain config must implement the [`OnChainConfig`] trait.
+    fn get_on_chain_config<C>(&self) -> Result<C>
+    where
+        Self: Sized,
+        C: OnChainConfig,
+    {
+        C::fetch_config(self).ok_or_else(|| {
+            anyhow!(
+                "failed to fetch on-chain config: {:?}",
+                std::any::type_name::<C>()
+            )
+        })
+    }
+    /// Modifies an on-chain config in-place.
+    ///
+    /// The Rust type representing the on-chain config must implement the [`OnChainConfig`] trait.
+    fn modify_on_chain_config<C, F>(&self, modify: F) -> Result<()>
+    where
+        Self: Sized,
+        C: OnChainConfig + Serialize,
+        F: FnOnce(&mut C) -> Result<()>,
+    {
+        let mut config = self.get_on_chain_config::<C>()?;
+        modify(&mut config)?;
+        self.set_on_chain_config(&config)
+    }
+
+    /// Stores a compiled Move module, with the module being already serialized into a blob.
+    fn add_module_blob(&self, module_id: &ModuleId, blob: impl Into<Bytes>) -> Result<()> {
+        self.set_state_value(
+            StateKey::module_id(module_id),
+            StateValue::new_legacy(blob.into()),
+        )
+    }
+    /// Stores a compiled Move module.
+    fn add_module(&self, module: &CompiledModule) -> Result<()> {
+        let mut blob = vec![];
+        module.serialize(&mut blob)?;
```

Comment:
Let's make sure this serializes things based on the config. I think here we need to
  - get `Features`
  - based on features get `aptos_prod_deserializer_config`
  - use the binary version from the config

Reply:
I don't think `.serialize` cares about the deserializer config -- it assumes the module it gets is valid. However I've now added an `fn get_module` and that implements the proper config-based deserialization as you mentioned.